bot/src/rpc_client_utils.rs

```diff
@@ -64,6 +65,14 @@ pub fn send_and_confirm_transactions(
                 },
             )?;
         } else {
+            // every 100 transactions, get a new blockhash
+            if num_transactions >= 100 {
+                info!("refreshing blockhash");
+                blockhash = rpc_client.get_recent_blockhash()?.0;
```

Comment:
Doesn't this break the blockhash expiration checks below since we now have multiple live blockhashes in play, but only account for the most recent

Reply:
Kind of -- I've been confused about that logic for a bit.

The blockhash check is done and transactions are resent *before* we've actually checked the transaction status.  Which means that you can take time to send 100 transactions, and if the blockhash expires right after that, you'll just resend them all without ever checking if the transactions even errored.  The resend logic should go after the check, and only if the status is `status.err.unwrap() == TransactionError::BlockhashNotFound`, no?

As I was thinking about better ways to do this, I thought: "oh, then we'd probably need a hashmap. Trent probably already thought that, which means that it's probably in his PR. I'll just read his PR" and it was *very* useful